sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -228,8 +228,11 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
     case DescribeColumn(_: ResolvedTable, _, _) =>
       throw new AnalysisException("Describing columns is not supported for v2 tables.")
 
-    case DropTable(catalog, ident, ifExists) =>
-      DropTableExec(catalog, ident, ifExists) :: Nil
+    case DropTable(r: ResolvedTable, ifExists, _) =>
+      DropTableExec(r.catalog, r.identifier, ifExists) :: Nil
+
+    case NoopDropTable(multipartIdentifier) =>
+      NoopDropTableExec(multipartIdentifier) :: Nil
```

Comment:
how about `LocalTableScanExec(Nil, Nil)`? It's only for EXPLAIN, so a logical plan is good enough. We don't need to create a real physical plan for it.

Reply:
Changed. The EXPLAIN output will look like the following:
```
|== Parsed Logical Plan ==
'DropTable true, false
+- 'UnresolvedTableOrView [testcat, db, notbl]

== Analyzed Logical Plan ==

NoopDropTable [testcat, db, notbl]

== Optimized Logical Plan ==
NoopDropTable [testcat, db, notbl]

== Physical Plan ==
LocalTableScan <empty>

```